### PR TITLE
Update entire mod to 1.11.2

### PR DESCRIPTION
**Pull request by PrinceOfAmber**
I used forge 1.11.2-13.20.0.2252

Overview of Changes:
Item stacks can never be null, instead there is the new ItemStack.EMPTY
mod id, item/block ids, and resource files had to be lowercased
many function names were changed
Entities are registered with a ResourceLocation along with a string & id
disabled Waila plugin

Changing the files from camelcase to lowercase was a bit tricky in git, i had to use git config --global core.ignorecase false and then  git add resources/assets/enderzoo/* -f

Dont think i missed anything

### CODE REVIEW

File: src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public boolean getCanSpawnHere() {
    */
   private boolean shouldAttackPlayer(EntityPlayer player) {
 
-    ItemStack itemstack = player.inventory.armorInventory[3];
+    ItemStack itemstack = player.inventory.armorInventory.get(3);
     if(itemstack != null && itemstack.getItem() == Item.getItemFromBlock(Blocks.PUMPKIN)) {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java]: here's a null check that can be dropped

File: src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java [hunk 2]
```diff
@@ -123,7 +123,7 @@ public boolean getCanSpawnHere() {
    */
   private boolean shouldAttackPlayer(EntityPlayer player) {
 
-    ItemStack itemstack = player.inventory.armorInventory[3];
+    ItemStack itemstack = player.inventory.armorInventory.get(3);
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java]: can you check if slot 3 is the correct one? I don't know if this was updated correctly after 1.7. Thx.

File: src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java [hunk 3]
```diff
@@ -70,10 +70,10 @@ protected boolean isMovementBlocked() {
   }
 
   @Override
-  public boolean processInteract(EntityPlayer player, EnumHand hand, ItemStack holding) {
+  public boolean processInteract(EntityPlayer player, EnumHand hand) {
     ItemStack itemstack = player.inventory.getCurrentItem();
-    if(itemstack != null && itemstack.getItem() == Items.SPAWN_EGG) {
-      return super.processInteract(player, hand, holding);
+    if(itemstack.isEmpty()==false && itemstack.getItem() == Items.SPAWN_EGG) {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java]: just drop the isEmpty() completely

File: src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java [hunk 4]
```diff
@@ -223,7 +223,7 @@ public boolean attackEntityAsMob(Entity target) {
   @Override
   public void writeEntityToNBT(NBTTagCompound root) {
     super.writeEntityToNBT(root);
-    if(armor != null) {
+    if(armor.isEmpty()==false) {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java]: `if(!armor.isEmpty()) {`

File: src/main/java/crazypants/enderzoo/item/ItemGuardiansBow.java [hunk 5]
```diff
@@ -110,15 +110,15 @@ public void onPlayerStoppedUsing(ItemStack stack, World worldIn, EntityLivingBas
...
 
       worldIn.playSound((EntityPlayer) null, entityplayer.posX, entityplayer.posY, entityplayer.posZ, SoundEvents.ENTITY_ARROW_SHOOT, SoundCategory.NEUTRAL,
           1.0F, 1.0F / (itemRand.nextFloat() * 0.4F + 1.2F) + drawRatio * 0.5F);
 
       if (!arrowIsInfinite) {
-        --itemstack.stackSize;
-        if (itemstack.stackSize == 0) {
+        itemstack.shrink(1);
+        if (itemstack.getCount() == 0) {
           entityplayer.inventory.deleteStack(itemstack);
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/item/ItemGuardiansBow.java]: have you checked with the normal bow's code that this should be called that way? The stack now isEmpty() and there's no way to find out what is was, so I don't see what that method could do with it.

File: src/main/java/crazypants/enderzoo/spawn/MobSpawnEventHandler.java [hunk 6]
```diff
@@ -213,12 +214,12 @@ public void onBlockHarvest(HarvestDropsEvent event) {
...
       direSlime.playLivingSound();
       for (ItemStack drop : event.getDrops()) {
         if (drop != null && drop.getItem() != null && drop.getItem() == Item.getItemFromBlock(Blocks.DIRT)) {
-          if (drop.stackSize > 1) {
-            drop.stackSize--;
+          if (drop.getCount() > 1) {
+            drop.shrink(1);
           } else if (event.getDrops().size() == 1) {
             event.getDrops().clear();
           } else {
```
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/spawn/MobSpawnEventHandler.java]: in isToolEffective(), line 235, you missed a stack null-check. (sry, cannot annotate there)

**Comment by HenryLoenwind** [comment]
@CrazyPants is this ok with you?

@PrinceOfAmber Have you had a look at the open issues? Maybe there's something that can be fixed together with the port.

FYI: I need to setup a new workspace to test this out on the weekend.

File: src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java (hunk 2 above)
Comment (PrinceOfAmber) [inline src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java]: Yeah ill check. do you remember what  slot it was looking for? pants or something? ill check the documentation too

File: src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java (hunk 3 above)
Comment (PrinceOfAmber) [inline src/main/java/crazypants/enderzoo/entity/EntityFallenMount.java]: good point, if it fails that second == spawn egg check the first is redundant

File: src/main/java/crazypants/enderzoo/item/ItemGuardiansBow.java (hunk 5 above)
Comment (PrinceOfAmber) [inline src/main/java/crazypants/enderzoo/item/ItemGuardiansBow.java]: I will double check tomorrow. isEmpty checks for the stack size == zero as well as a few other things.  I could replace the count==0 to isempty call. this is from source

`    {
        return this == EMPTY ? true : (getItemRaw() != null && getItemRaw() != Item.getItemFromBlock(Blocks.AIR) ? (this.stackSize <= 0 ? true : this.itemDamage < -32768 || this.itemDamage > 65535) : true);
    }`

**Comment by PrinceOfAmber** [comment]
Thanks for the review! I will update code with your feedback sometime tomorrow.  After that ill check out issues.

File: src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java (hunk 2 above)
Comment (PrinceOfAmber) [inline src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java]: Yep so its //    3: Helmet, 2: Chestpiece, 1: Legs, 0: Boots

Also i checked EntityEnderman.class and it says the same thing

`ItemStack itemstack = (ItemStack)player.inventory.armorInventory.get(3);

        if (itemstack.getItem() == Item.getItemFromBlock(Blocks.PUMPKIN))
        {
            return false;
        }`

**Comment by PrinceOfAmber** [comment]
Updated based on feedback.

Also found one bug in the WitherWitch thrown potion , it says

[08:00:18] [Server thread/ERROR]: ThrownPotion entity 1561 has no item?!

so ill have to review BrewingUtils a bit

**Comment by PrinceOfAmber** [comment]
Found an issue with my changes to "addVelocity" vs "moveEntity" , looks like the owls cant/wont move and are always frozen in place, looking into it

File: src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java (hunk 2 above)
Comment (HenryLoenwind) [inline src/main/java/crazypants/enderzoo/entity/EntityEnderminy.java]: so it's correct. thx for checking.

**Comment by PrinceOfAmber** [comment]
its quite stable now, and ive addressed a few of the issues from the tracker as well. 

What do you think?
@HenryLoenwind  @CrazyPants  @SleepyTrousers

**Comment by CrazyPants** [comment]
I have just seen this PR. Thanks so much for updating the mod. I haven't
had a chance to look at the changes but I am happy for them to be merged as
I won't be able to look at them for at least a few weeks.

**Comment by PrinceOfAmber** [comment]
Cool thats awesome!  Looking forward to the merge.

If you are busy, do you want me to send you a built jar file for you to release (on curseforge)?

IFor contact  am at https://minecraft.curseforge.com/members/Lothrazar and https://twitter.com/lothrazar

**Comment by CrazyPants** [comment]
Is this good to merge and release as is?

**Comment by PrinceOfAmber** [comment]
Should be good, been using it in single player with other mods.  DM me on twitter anytime

**Pull request merged**